components/helm-broker/internal/controller/broker/broker_syncer.go

```diff
@@ -0,0 +1,130 @@
+	clusterServiceBrokersGetter v1beta12.ClusterServiceBrokersGetter
+	serviceBrokerGetter         v1beta12.ServiceBrokersGetter
+	clusterBrokerName           string
+	log                         logrus.FieldLogger
+}
+
+// NewServiceBrokerSyncer allows to sync the ServiceBroker.
+func NewServiceBrokerSyncer(clusterServiceBrokersGetter v1beta12.ClusterServiceBrokersGetter, serviceBrokerGetter v1beta12.ServiceBrokersGetter, clusterBrokerName string, log logrus.FieldLogger) *ServiceBrokerSyncer {
+	return &ServiceBrokerSyncer{
+		clusterServiceBrokersGetter: clusterServiceBrokersGetter,
+		clusterBrokerName:           clusterBrokerName,
+		serviceBrokerGetter:         serviceBrokerGetter,
+		log:                         log.WithField("service", "clusterservicebroker-syncer"),
+	}
+}
+
+const maxSyncRetries = 5
+
+// Sync syncs the ServiceBrokers, does not fail if the broker does not exists
+func (r *ServiceBrokerSyncer) Sync() error {
+	r.log.Infof("Trigger Service Catalog to refresh ClusterServiceBroker %s", r.clusterBrokerName)
+	for i := 0; i < maxSyncRetries; i++ {
+		broker, err := r.clusterServiceBrokersGetter.ClusterServiceBrokers().Get(r.clusterBrokerName, v1.GetOptions{})
+		switch {
+		// do not return error if the broker does not exists, the method is dedicated to update
+		// ClusterServiceBroker resource if exists. If it is not created yet
+		// - it will be created in the future and Service Catalog will call the 'catalog' endpoint soon.
+		case apiErrors.IsNotFound(err):
+			return nil
+		case err != nil:
+			return errors.Wrapf(err, "while getting ClusterServiceBrokers %s", r.clusterBrokerName)
+		}
+
+		// update RelistRequests to trigger the relist
+		broker.Spec.RelistRequests = broker.Spec.RelistRequests + 1
+
+		_, err = r.clusterServiceBrokersGetter.ClusterServiceBrokers().Update(broker)
+		switch {
+		case err == nil:
+			return nil
+		case apiErrors.IsConflict(err):
+			r.log.Infof("(%d/%d) ClusterServiceBroker %s update conflict occurred.", i, maxSyncRetries, broker.Name)
+		case err != nil:
+			return errors.Wrapf(err, "while updating ClusterServiceBroker %s", broker.Name)
+		}
+	}
+
+	return fmt.Errorf("could not sync cluster service broker (%s) after %d retries", r.clusterBrokerName, maxSyncRetries)
+}
+
+// SyncServiceBroker syncing the helm-broker ns-broker in the given namespace
+func (r *ServiceBrokerSyncer) SyncServiceBroker(namespace string) error {
+	brokerClient := r.serviceBrokerGetter.ServiceBrokers(namespace)
+
+	for i := 0; i < maxSyncRetries; i++ {
+		broker, err := brokerClient.Get(NamespacedBrokerName, v1.GetOptions{})
+		if err != nil {
+			return errors.Wrapf(err, "while getting ServiceBroker %q [namespace: %s]", NamespacedBrokerName, namespace)
+		}
+
+		broker.Spec.RelistRequests = broker.Spec.RelistRequests + 1
+
+		_, err = brokerClient.Update(broker)
+		if err == nil {
+			return nil
+		}
+		if !apiErrors.IsConflict(err) {
+			return fmt.Errorf("could not sync service broker (%s)", err)
```

Comment:
Do not return from the method because next retry never happen

Reply:
done